ignite/handlers/lr_finder.py

```diff
@@ -118,13 +118,25 @@ def _run(
             )
 
         self.logger.debug(f"Running LR finder for {num_iter} iterations")
-        if start_lr is None:
-            start_lr = optimizer.param_groups[0]["lr"]
         # Initialize the proper learning rate policy
-        if step_mode.lower() == "exp":
-            start_lr_list = [start_lr] * len(optimizer.param_groups)
-            self._lr_schedule = LRScheduler(_ExponentialLR(optimizer, start_lr_list, end_lr, num_iter))
+        if step_mode.lower() == "exp":       
+            if start_lr is None:
+                start_lr_list = [optimizer.param_groups[i]["lr"] for i in range(len(optimizer.param_groups))]
+            elif isinstance(start_lr, float):
+                start_lr_list = [start_lr] * len(optimizer.param_groups)
+            else:
+                assert len(start_lr) == len(optimizer.param_groups)
+                start_lr_list = start_lr
+            if isinstance(end_lr, float):
+                end_lr_list = [end_lr] * len(optimizer.param_groups)
+            else:
+                assert len(end_lr) == len(optimizer.param_groups)
```

Comment:
We should replace `assert condition` with `if not condition: raise ValueError / TypeError`. Please check this file for an example